File: demo-pojo/src/main/java/com/demo/entity/Order.java
package com.demo.entity;

import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 订单实体，对应 orders 表
 */
@Data
public class Order {

    private Long id;            // 主键ID
    private String orderNo;     // 订单号（业务展示用）

    private Long buyerId;       // 买家ID
    private Long sellerId;      // 卖家ID

    private BigDecimal totalAmount;   // 订单总金额（所有明细合计）

    /**
     * 订单状态（数据库存小写字符串）：
     * pending   - 待付款
     * paid      - 已付款待发货
     * shipped   - 已发货待收货
     * completed - 已完成
     * cancelled - 已取消
     *
     * 业务代码中建议通过 OrderStatus 枚举使用：
     *   - 落库：order.setStatus(OrderStatus.PAID.getDbValue());
     *   - 读库：OrderStatus.fromDbValue(order.getStatus());
     */
    private String status;

    // 收货信息
    private String shippingAddress;   // 收货地址快照
    private String shippingCompany;   // 物流公司（新加）
    private String trackingNo;        // 运单号（新加）
    private String shippingRemark;    // 发货备注（可选，新加）

    // 时间信息
    private LocalDateTime createTime;   // 下单时间
    private LocalDateTime payTime;      // 支付时间
    private LocalDateTime completeTime; // 完成时间（确认收货等）
    private LocalDateTime updateTime;   // 最后更新时间（新加，可选）
}

File: demo-admin-ui/src/api/order.ts
import request from '@/utils/request'

export type AdminOrderSortField = 'createTime' | 'payTime'
export type AdminOrderSortOrder = 'asc' | 'desc'

export interface AdminOrderItem {
  orderId: number
  orderNo: string
  buyerId: number
  buyerNickname: string
  buyerMobile: string
  sellerId: number
  sellerNickname: string
  sellerMobile: string
  productId: number | null
  productTitle: string | null
  productThumbnail: string | null
  flagCount: number
  totalAmount: number | string
  status: string
  shippingCompany: string | null
  trackingNo: string | null
  createTime: string | null
  payTime: string | null
  shipTime: string | null
  completeTime: string | null
  cancelTime: string | null
  cancelReason: string | null
}

export interface AdminOrderPage {
  list: AdminOrderItem[]
  total: number
  page: number
  pageSize: number
}

export interface AdminOrderQuery {
  page: number
  pageSize: number
  keyword?: string
  status?: string
  startTime?: string
  endTime?: string
  sortField: AdminOrderSortField
  sortOrder: AdminOrderSortOrder
}

export interface AdminOrderLineItem {
  productId: number
  productTitle: string
  productThumbnail: string | null
  productStatus: string
  price: number | string
  quantity: number
}

export interface AdminOrderFlag {
  id: number
  type: string
  remark: string | null
  createdBy: number | null
  createdByNickname: string | null
  createTime: string | null
}

export interface AdminOrderDetail extends Omit<AdminOrderItem, 'productId' | 'productTitle' | 'productThumbnail' | 'flagCount'> {
  shippingAddress: string | null
  shippingRemark: string | null
  afterSaleId: number | null
  afterSaleStatus: string | null
  afterSaleReason: string | null
  afterSalePlatformRemark: string | null
  items: AdminOrderLineItem[]
  flags: AdminOrderFlag[]
}

export type OrderFlagType =
  | 'PAYMENT_RISK'
  | 'PRICE_ANOMALY'
  | 'DELIVERY_RISK'
  | 'AFTERSALE_RISK'
  | 'ACCOUNT_RISK'
  | 'MANUAL_REVIEW'

export interface CreateOrderFlagPayload {
  type: OrderFlagType
  remark?: string
}

const statusLabels: Record<string, string> = {
  pending: '待支付',
  paid: '待发货',
  shipped: '待收货',
  completed: '已完成',
  cancelled: '已取消',
}

const flagLabels: Record<OrderFlagType, string> = {
  PAYMENT_RISK: '支付异常',
  PRICE_ANOMALY: '金额异常',
  DELIVERY_RISK: '发货风险',
  AFTERSALE_RISK: '售后风险',
  ACCOUNT_RISK: '账号风险',
  MANUAL_REVIEW: '人工复核',
}

export function getAdminOrderStatusLabel(status: string) {
  return statusLabels[status] || `未知状态：${status}`
}

export function getOrderFlagLabel(type: string) {
  return flagLabels[type as OrderFlagType] || `未知标记：${type}`
}

export function getAdminOrders(query: AdminOrderQuery) {
  return request.get<any, AdminOrderPage>('/admin/orders', { params: query })
}

export function getAdminOrderDetail(orderId: number) {
  return request.get<any, AdminOrderDetail>(`/admin/orders/${orderId}`)
}

export function getAdminOrderFlags(orderId: number) {
  return request.get<any, AdminOrderFlag[]>(`/admin/orders/${orderId}/flags`)
}

export function createAdminOrderFlag(orderId: number, payload: CreateOrderFlagPayload) {
  return request.post<any, string>(`/admin/orders/${orderId}/flags`, payload)
}
